src/components/Autocomplete/Autocomplete.tsx

```diff
@@ -0,0 +1,101 @@
+import { FC, useState, useEffect } from 'react';
+import cn from 'classnames';
+import { debounce } from '../../helper';
+import { Person } from '../../types/Person';
+import './Autocomplete.scss';
+
+type Props = {
+  options: Person[],
+  getValue: (value: Person) => void;
+};
+
+const getCurrentOptions = (options: Person[], query: string): Person[] | [] => {
+  if (query) {
+    return options.filter(({ name }) => {
+      return name.toLowerCase().includes(query.toLowerCase());
+    });
+  }
+
+  return [...options];
+};
+
+export const Autocomplete: FC<Props> = ({ options, getValue }) => {
+  const [showMenu, setShowMenu] = useState(false);
+  const [currentValue, setCurrentValue] = useState('');
+  const [searchQuery, setSearchQuery] = useState('');
+  const currentOptions = getCurrentOptions(options, searchQuery);
+
+  useEffect(() => {
+    getValue(currentOptions.find(({ name }) => {
+      return name === currentValue;
+    }) as Person);
```

Comment:
```suggestion
    getValue(currentOptions.find(({ name }) => {
      return name === currentValue;
    }) || null);
```
Avoid `as`, in accordance with your state `const [person, setPerson] = useState<Person | null>(null);` person could be null, so path null in this case.